src/commcare_cloud/commands/ansible/ops_tool.py

```diff
@@ -487,50 +486,83 @@ class AuditEnvironment(_AnsiblePlaybookAlias):
         self.env_info_dict = {}
+        # We set the audits_directory in the `run` method
+        self.audits_directory = os.path.expanduser(f"~/.commcare-cloud/audits")
+        self.environment = ""
+        self.curr_audit_directory = ""
 
     def run(self, args, unknown_args):
+        self.args = args
         self.environment = get_environment(args.env_name)
         self.env_info_dict["environment"] = args.env_name
-        self.collect_commcare_cloud_details()
-        self.collect_commcare_hq_details()
-        self.validate_environment_settings()
-    
-        self._create_audit_dir()
-
-    def _create_audit_dir(self):
-        env_name = self.env_info_dict["environment"]
-        audit_name = datetime.utcnow().strftime(DATE_FMT)
         
-        self.audits_directory = os.path.expanduser(f"~/.commcare-cloud/audits/{env_name}/{audit_name}")
-        if not os.path.isdir(self.audits_directory):
-            os.makedirs(self.audits_directory)
+        env_name = args.env_name
+        audit_name = datetime.utcnow().strftime('%Y-%m-%d_%H.%M.%S')
+        self.curr_audit_directory = os.path.join(self.audits_directory, f"{env_name}/{audit_name}")
+        self._ensure_dir_exists(self.curr_audit_directory)
 
-        file_path = os.path.join(self.audits_directory, "info.json")
+        self._collect_commcare_cloud_details()
+        self._collect_commcare_hq_details()
+        self._validate_environment_settings()
+        self._collect_control_machine_os_level_info()
+        self._collect_service_folder_permissions_info()
+
+        self._write_details()
+
+    def _ensure_dir_exists(self, path):
+        if not os.path.isdir(path):
+            os.makedirs(path)
+
+    def _write_details(self):
+        file_path = os.path.join(self.curr_audit_directory, "info.json")
         with open(file_path, "w") as file:
             json.dump(self.env_info_dict, fp=file)
 
-    def collect_commcare_cloud_details(self):
+    def _collect_commcare_cloud_details(self):
         repo = git.Repo(search_parent_directories=True)
         hexsha = repo.head.object.hexsha
         self.env_info_dict["commcare_cloud"] = {"commit": hexsha}
 
-    def collect_commcare_hq_details(self):
+    def _collect_commcare_hq_details(self):
         hexsha = get_deployed_version(self.environment)
         self.env_info_dict["commcare_hq"] = {"commit": hexsha}
 
-    def validate_environment_settings(self):
+    def _validate_environment_settings(self):
         settings_validaton = {"passed": True, "failure_reason": None}
         try:
             self.environment.check()
         except Exception as exception:
             settings_validaton["passed"] = False
             settings_validaton["failure_reason"] = str(exception)
         self.env_info_dict["settings_validation"] = settings_validaton
+
+    def _collect_control_machine_os_level_info(self):
+        os_data = {}
+        
+        os_distrib = {}
+        with open("/etc/lsb-release", "r") as file:
```

Comment:
Can't this be captured in `commcare-audit.yml` as well?

Reply:
Yes you are correct. I'm going to create a separate PR for this change/refactor